filebeat/harvester/reader/line.go

```diff
@@ -90,12 +98,22 @@ func (l *Line) Next() ([]byte, int, error) {
 	return bytes, sz, nil
 }
 
+func (r *Line) Close() error {
+	close(r.done)
+	return r.rawInput.Close()
```

Comment:
What do you mean by the second point? What you say is that we should keep `r.rawInput.Close()` here and not have it done by the harvester? I did some quick tests and it doesn't seem to be an issue to call close multiple times.